CoreFoundation/AppServices.subproj/CFUserNotification.h

```diff
@@ -131,53 +131,53 @@ CF_INLINE CFOptionFlags CFUserNotificationPopUpSelection(CFIndex n) {return ((CF
 /* Keys */
 
 CF_EXPORT
-const CFStringRef kCFUserNotificationIconURLKey;
+extern const CFStringRef kCFUserNotificationIconURLKey;
```

Comment:
`CF_EXPORT` _must_ imply `extern`. If the current definition doesn't, we must change its `#define`.

Reply:
The current definition _does_ imply it:

```C
#if TARGET_OS_WIN32
    #if !defined(CF_EXPORT)
        #if defined(CF_BUILDING_CF) && defined(__cplusplus)
            #define CF_EXPORT extern "C" __declspec(dllexport)
        #elif defined(CF_BUILDING_CF) && !defined(__cplusplus)
            #define CF_EXPORT extern __declspec(dllexport)
        #elif defined(__cplusplus)
            #define CF_EXPORT extern "C" __declspec(dllimport)
        #else
            #define CF_EXPORT extern __declspec(dllimport)
        #endif
    #endif
#else
#define CF_EXPORT extern
#endif
```

Reply:
`CF_EXPORT` really shouldn't imply `extern`.  Consider the following bit in the implementation:

```
CF_EXPORT void CFFunction(void) {
}
```

The definition itself should not get marked as `extern`.  I think that this is why this was mixed into the other change.  We should split the two apart.  There is a fair amount of usage of this pattern through out the current source tree.

Reply:
The function is incorrect in this instance. We should split that into:

```
CF_EXTERN void CFFunction(void);
void CFFunction() { … }
```

Reply:
@compnerd is there any platform we want to build on that doesn't support extern on function declarations? I just tested with clang on Linux and it does support it.

Reply:
(by which I mean functions with bodies.)

Reply:
Windows :-)

Reply:
Even on clang?

Reply:
Yeah, `clang-cl` matches the behaviour